components/style/stylist.rs

```diff
@@ -1366,9 +1366,20 @@ impl SelectorMap<Rule> {
 
         let mut rules_list = vec![];
         for rule in self.other.iter() {
-            if rule.selector.inner.complex.iter_raw().next().is_none() {
-                rules_list.push(rule.to_applicable_declaration_block(cascade_level));
+            let mut iter = rule.selector.inner.complex.iter_raw();
+            match iter.next() {
```

Comment:
can you move it to a method in `Selector`? I'm also tweaking this code for the pseudo-element things. Also, it might look cleaner that way.